### PR TITLE
feat: enforce markdownlint in the repository

### PR DESCRIPTION
**Pull request by julienrbrt**
Follow-up of  https://github.com/cosmos/sdk-tutorials/pull/1164#discussion_r919996384

The `.markdownlint.json` configuration has been introduced, but is not yet enforced.
This PR aims to enforce the rule in all the files after having discussed the rules present in `.markdownlint.json` (so not all files have been touched yet).

The list of available rules is available here: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md

### Change scope

The changes in this Pull Request include (please tick all that apply):

- [ ] Small language/grammar fixes
- [ ] Small content fixes 
- [ ] Addition of new content
- [ ] Sample code/command updates
- [ ] Platform fixes
- [ ] Other

### CODE REVIEW

File: .markdownlint.json [hunk 1]
```diff
@@ -3,10 +3,12 @@
     "MD004": {"style": "asterisk"},
     "MD007": { "indent": 4 },
     "MD013": false,
+    "MD014": false,
```
Comment (xavierlepretre) [inline .markdownlint.json:6]: @coldice This one is about [`$` in code blocks](MD014). Not sure what `false` means here.
@julienrbrt The rationale mentioned is that it is easier to copy without the `$` but our copy button removes them.
I am in favour of having the `$`.

File: .markdownlint.json [hunk 2]
```diff
@@ -3,10 +3,12 @@
...
     "MD024": { "siblings_only": true },
     "MD025": false,
     "MD033": false,
     "MD034": false,
+    "MD049": {"style": "underscore"},
```
Comment (xavierlepretre) [inline .markdownlint.json:11]: This one is for emphasis (italics), we have written them with `_` throughout so we ought to make it official.

File: .vscode/extensions.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "recommendations": ["DavidAnson.vscode-markdownlint"]
```
Comment (xavierlepretre) [inline .vscode/extensions.json:2]: @coldice Like this everyone is prompted to install it.

File: .markdownlint.json (hunk 1 above)
Comment (julienrbrt) [reply .markdownlint.json:6]: The `false` disable the rule :)

Comment (xavierlepretre) [reply .markdownlint.json:6]: So there is no way to enforce the presence of `$`? Only a way to enforce (or not) the absence of `$`?

Comment (julienrbrt) [reply .markdownlint.json:6]: Nope this rule can only be enabled of disabled

**Review by xavierlepretre — APPROVED** [review]

**Pull request merged**